lib/util/subprocess.py

```diff
@@ -3,21 +3,61 @@
 from lib import util
 
 
+class VerboseCalledProcessError(subprocess.CalledProcessError):
+    """
+    A CalledProcessError that includes captured stderr in its string representation.
+    """
+    def __str__(self):
+        base = super().__str__()
+        if self.stderr:
+            stderr_output = self.stderr
+            if isinstance(stderr_output, bytes):
+                stderr_output = stderr_output.decode(errors='replace')
+            # truncate very long stderr to keep error messages readable
+            max_len = 2000
+            if len(stderr_output) > max_len:
+                stderr_output = stderr_output[:max_len] + '\n... (truncated)'
+            return f"{base} \nCaptured stderr: \n{stderr_output}"
+        return base
+
+    def __repr__(self):
+        return self.__str__()
+
+
 def _format_subprocess_cmd(cmd):
     if isinstance(cmd, (list, tuple)):
         return ' '.join(str(x) for x in cmd)
     else:
         return cmd
 
 
-def subprocess_run(cmd, *, skip_frames=0, **kwargs):
+def subprocess_run(cmd, *, check=False, skip_frames=0, stderr=None, **kwargs):
     """
     A simple wrapper for the real subprocess.run() that logs the command used.
+
+    When stderr=subprocess.PIPE and check=True, automatically converts
+    CalledProcessError to VerboseCalledProcessError with captured stderr details.
     """
     # when logging, skip current stack frame - report the place we were called
     # from, not util.subprocess_run itself
     util.log(f'running: {_format_subprocess_cmd(cmd)}', skip_frames=skip_frames+1)
-    return subprocess.run(cmd, **kwargs)
+
+    use_verbose_errors = (stderr == subprocess.PIPE and check)
+
+    try:
+        return subprocess.run(cmd, stderr=stderr, **kwargs)
+    except subprocess.CalledProcessError as e:
+        if use_verbose_errors and e.stderr is not None:
+            # re-raise as VerboseCalledProcessError with stderr details
+            raise VerboseCalledProcessError(
+                returncode=e.returncode,
+                cmd=e.cmd,
+                output=e.output,
```

Comment:
```suggestion
                stdout=e.stdout,
```
both are the same, but for consistency with `stderr`, using `stdout` seems more appropriate.